.github/workflows/e2e.yml

```
@@ -0,0 +1,275 @@
+
+      - name: Install PyTorch
+        run: |
+          pip install pytorch/dist/*.whl
+          python -c "import torch;print(torch.__version__)"
+
+      - name: Save PyTorch to a cache
+        if: ${{ steps.pytorch-cache.outputs.status == 'miss' }}
+        uses: ./.github/actions/save
+        with:
+          path: ${{ steps.pytorch-cache.outputs.path }}
+          dest: ${{ steps.pytorch-cache.outputs.dest }}
+
+      - name: Load IPEX wheels from a cache
+        id: ipex-cache
+        uses: ./.github/actions/load
+        with:
+          path: intel-extension-for-pytorch/dist
+          key: ipex-$PYTHON_VERSION-$IPEX_COMMIT_ID
+
+      - name: Build IPEX
+        if: ${{ steps.ipex-cache.outputs.status == 'miss' }}
+        run: |
+          source ~/intel/oneapi/setvars.sh
+          export USE_AOT_DEVLIST='pvc'
+          git clone --single-branch -b $IPEX_BRANCH --recurse-submodules --jobs 8 $IPEX_REPO
+          cd intel-extension-for-pytorch
+          pip install -r requirements.txt
+          python setup.py bdist_wheel
+
+      - name: Install IPEX
+        run: |
+          source ~/intel/oneapi/setvars.sh
+          pip install intel-extension-for-pytorch/dist/*.whl
+          python -c "import torch;import intel_extension_for_pytorch as ipex;print(ipex.__version__)"
+
+      - name: Save IPEX wheels to a cache
+        if: ${{ steps.ipex-cache.outputs.status == 'miss' }}
+        uses: ./.github/actions/save
+        with:
+          path: ${{ steps.ipex-cache.outputs.path }}
+          dest: ${{ steps.ipex-cache.outputs.dest }}
+
+      - name: Generate Triton cache key
+        id: triton-key
+        run: |
+          COMPOSITE_KEY=$(echo $PYTHON_VERSION $LLVM_COMMIT_ID $GITHUB_SHA | sha256sum - | cut -d\  -f1)
+          echo "key=triton-$COMPOSITE_KEY" >> $GITHUB_OUTPUT
+
+      - name: Load Triton wheels from a cache
+        id: triton-cache
+        uses: ./.github/actions/load
+        with:
+          path: python/dist
+          key: ${{ steps.triton-key.outputs.key }}
+
+      - name: Generate packages cache key
+        if: ${{ steps.triton-cache.outputs.status == 'miss' }}
+        id: packages-key
+        env:
+          # Increase this value to reset cache
+          CACHE_NUMBER: 1
+        run: |
+          COMPOSITE_KEY=$(echo $LLVM_COMMIT_ID ${{ hashFiles('scripts/compile-triton.sh') }} | sha256sum - | cut -d\  -f1)
+          echo "key=packages-$COMPOSITE_KEY-$CACHE_NUMBER" >> $GITHUB_OUTPUT
+
+      - name: Load packages from a cache
+        if: ${{ steps.triton-cache.outputs.status == 'miss' }}
+        id: packages-cache
+        uses: ./.github/actions/load
+        with:
+          path: $HOME/packages
+          key: ${{ steps.packages-key.outputs.key }}
+
+      - name: Build packages
+        if: ${{ steps.triton-cache.outputs.status == 'miss' && steps.packages-cache.outputs.status == 'miss' }}
+        run: |
+          ./scripts/compile-triton.sh --skip-triton
```

Note: `skip-triton` option will be unable after https://github.com/intel/intel-xpu-backend-for-triton/pull/480/files.
```suggestion
          ./scripts/compile-triton.sh --llvm
```

---

Done